src/datareader.cpp

```diff
@@ -19,12 +20,15 @@
 #include <xmfile.h>
 
 
+#include <QCryptographicHash>
```

Comment:
As far as I can see, this is not use in the new code. Same for `QImage`, correct me if I'm wrong.